spec/support/cart_page.rb

```diff
@@ -30,7 +30,7 @@ def visit_page
 
   def add_position_quantity(position_id = @positions.first.id)
     within("#pos#{position_id}") do
-      sleep 0.3
+      sleep 0.4
```

Comment:
Using 'sleep' in tests is a highly dangerous antipattern! You already have increased a sleep timer to make tests pass - and that's because it is dependent on a lot of factors: computing power of machine running tests, whether or not garbage collector started, background processes running or not, network latency ( not in this case, but you get the point ).
Either use wait for ajax or try increasing a bit capybara's wait time ( when it uses finders it waits for some default time before raising 'element not found' exception )